test/e2e/util/fixture.go

```diff
@@ -301,13 +304,33 @@ func validateAWSGuestResourcesDeletedFunc(ctx context.Context, t *testing.T, inf
 		})
 		if err != nil {
 			t.Logf("WARNING: failed to list resources by tag: %v. Not verifying cluster is cleaned up.", err)
-		} else if hasGuestResources(t, output.ResourceTagMappingList) {
-			t.Errorf("FAIL: found %d remaining resources for guest cluster", len(output.ResourceTagMappingList))
-			for i := 0; i < len(output.ResourceTagMappingList); i++ {
-				t.Logf("Resource: %s, tags: %s", awssdk.StringValue(output.ResourceTagMappingList[i].ResourceARN), resourceTags(output.ResourceTagMappingList[i].Tags))
+			return
+		}
+
+		for _, mapping := range output.ResourceTagMappingList {
+			resourceARN, err := arn.Parse(awssdk.StringValue(mapping.ResourceARN))
+			if err != nil {
+				t.Logf("WARNING: failed to parse ARN %s", awssdk.StringValue(mapping.ResourceARN))
+				continue
 			}
-		} else {
-			t.Log("SUCCESS: found no remaining guest resources")
+
+			t.Logf("Found resource: %s, tags: %s, service: %s", resourceARN, resourceTags(mapping.Tags), resourceARN.Service)
+			if resourceARN.Service == "elb" {
+				t.Logf("Waiting for resource: %s, tags: %s to be deleted", resourceARN, resourceTags(mapping.Tags))
+				elbClient := elbv2.New(awsSession, awsConfig)
```

Comment:
The loadbalancer may be legacy (elb) or nlb. We need to use logic like [the one in the installer](https://github.com/openshift/installer/blob/75173a17cf68af8015b40b725eb2400e9737d6b9/pkg/destroy/aws/elbhelpers.go#L25-L38) to determine which one it is and then use the appropriate client (elb.New() for elbs, elbv2.New() for nlbs)